executor/analyze.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"math/rand"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/meta"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/plan/statistics"
+	"github.com/pingcap/tidb/plan/statscache"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// AnalyzeExec represents Analyze executor.
+type AnalyzeExec struct {
+	schema     expression.Schema
+	tblInfo    *model.TableInfo
+	ctx        context.Context
+	indOffsets []int
+	colOffsets []int
+	pkOffset   int
+	Srcs       []Executor
+}
+
+const (
+	maxSampleCount     = 10000
+	defaultBucketCount = 256
+)
+
+// Schema implements the Executor Schema interface.
+func (e *AnalyzeExec) Schema() expression.Schema {
+	return e.schema
+}
+
+// Close implements the Executor Close interface.
+func (e *AnalyzeExec) Close() error {
+	for _, src := range e.Srcs {
+		err := src.Close()
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// Next implements the Executor Next interface.
+func (e *AnalyzeExec) Next() (*Row, error) {
+	for _, src := range e.Srcs {
+		ae := src.(*AnalyzeExec)
```

Comment:
If add an executor like `AnalyzeDriverExec`, it means I should also add something like `AnalyzeDriver` in the planner, because if not, the logic for build executor from plan will be still like this. Does it really worth?